src/query/service/src/table_functions/read_parquet/table.rs

```diff
@@ -0,0 +1,226 @@
+        }
+
+        let file_metas = file_paths
+            .iter()
+            .map(|path| {
+                let file_meta = read_parquet_meta(path)?;
+                Ok(ParquetFileMeta {
+                    location: path.to_string_lossy().to_string(),
+                    file_meta,
+                })
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        // Infer schema from the first parquet file.
+        // Assume all parquet files have the same schema.
+        // If not, throw error during reading.
+        let schema = infer_schema(&file_metas[0].file_meta);
+
+        let table_info = TableInfo {
+            ident: TableIdent::new(table_id, 0),
+            desc: format!("'{}'.'{}'", database_name, table_func_name),
+            name: table_func_name.to_string(),
+            meta: TableMeta {
+                schema: Arc::new(schema),
+                engine: "SystemReadParquet".to_string(),
+                // Assuming that created_on is unnecessary for function table,
+                // we could make created_on fixed to pass test_shuffle_action_try_into.
+                created_on: Utc.from_utc_datetime(&NaiveDateTime::from_timestamp(0, 0)),
+                updated_on: Utc.from_utc_datetime(&NaiveDateTime::from_timestamp(0, 0)),
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+
+        let mut builder = opendal::services::fs::Builder::default();
+        builder.root("/");
+        let operator = Operator::new(builder.build()?);
+
+        Ok(Arc::new(ParquetTable {
+            table_args,
+            table_info,
+            file_metas,
+            operator,
+        }))
+    }
+}
+
+#[async_trait::async_trait]
+impl Table for ParquetTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    fn benefit_column_prune(&self) -> bool {
+        true
+    }
+
+    fn support_prewhere(&self) -> bool {
+        true
+    }
+
+    fn has_exact_total_row_count(&self) -> bool {
+        true
+    }
+
+    fn table_args(&self) -> Option<Vec<DataValue>> {
+        Some(self.table_args.clone())
+    }
+
+    async fn read_partitions(
+        &self,
+        _ctx: Arc<dyn TableContext>,
+        push_down: Option<PushDownInfo>,
+    ) -> Result<(PartStatistics, Partitions)> {
+        self.do_read_partitions(push_down)
+    }
+
+    fn read_data(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        plan: &DataSourcePlan,
+        pipeline: &mut Pipeline,
+    ) -> Result<()> {
+        self.do_read_data(ctx, plan, pipeline)
+    }
+}
+
+impl TableFunction for ParquetTable {
+    fn function_name(&self) -> &str {
+        self.name()
+    }
+
+    fn as_table<'a>(self: Arc<Self>) -> Arc<dyn Table + 'a>
+    where Self: 'a {
+        self
+    }
+}
+
+fn read_parquet_meta(file: &PathBuf) -> Result<FileMetaData> {
+    let mut file = File::open(file).map_err(|e| {
```

Comment:
So we only support reading local files?

Reply:
`read_parquet` is only used for local files.

Reply:
It aims to help Databend developers to do some performance tests.

Reply:
> `read_parquet` is only used for local files.

If so, we need to disable it by default and only enable by config.

Reply:
`COPY` controls this behavior by https://github.com/datafuselabs/databend/blob/main/src/query/config/src/outer_v0.rs#L254-L255

Reply:
> `COPY` controls this behavior by [`main`/src/query/config/src/outer_v0.rs#L254-L255](https://github.com/datafuselabs/databend/blob/main/src/query/config/src/outer_v0.rs?rgh-link-date=2022-12-04T08%3A35%3A03Z#L254-L255)

I think we can also use this here.

Reply:
But it seems we can't get the context when create table functions.

Reply:
> I expect we can read files from any location we supported in COPY. BTW, reading from local files should be disabled in cloud services.

We can support local files for this pr and stage files later, like:

```
read_parquet('@stage_name/a/b/*.parquet')
```

Reply:
> But it seems we can't get the context when create table functions.

can we just validate it during query?

Reply:
Yes. But we need to read meta when create the function. This is means, we will read the local files at the most begin.

Reply:
We should not read files during create.

Reply:
Got.